urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Nope. =) `Response` will handle this appropriately. In fact, we can even remove the explicit call to `release_conn` here as `Response` will do that automatically once the entire body is read.